src/gdsync.cpp

```diff
@@ -452,15 +459,44 @@ static int gds_fill_poll(gds_op_list_t &ops, CUdeviceptr ptr, uint32_t magic, in
                 retcode = EINVAL;
                 goto out;
         }
+#if 0        
         if (need_flush)
                 param.waitValue.flags |= CU_STREAM_WAIT_VALUE_FLUSH;
+#endif
         gds_dbg("op=%d addr=%p value=%08x cond=%s flags=%08x\n",
                 param.operation,
                 (void*)param.waitValue.address,
                 param.waitValue.value,
                 cond_str,
                 param.waitValue.flags);
+
         ops.push_back(param);
+
+        //Fake wait value on 0 gmemory area
+        if(need_flush && gds_flusher_enabled())
```

Comment:
this code assumes that the current GPU context is the same as the one associated to the stream in which this ops will be pushed to.

Reply:
this is a bug which deserves a new issue